src/ast/codegen_helper.cpp

```diff
@@ -0,0 +1,25 @@
+#include "codegen_helper.h"
+
+namespace bpftrace::ast {
+
+/* needAssignMapStatementAllocation determines if a map assignment requires
```

Comment:
Hope you don't mind. Here is a rewrite of this comment - more for the sanity of my future self:
"needAssignMapStatementAllocation determines if a map assignment requires a new memory allocation. This happens only in a few cases e.g. if there are two map assignments of tuples of different sizes (`@x = ("xxx", 1); @x = ("xxxxxxx", 1);` which requires a 0 memsetting and copying of each tuple element into the new allocation before calling `bpf_map_update_elem`.

Another case when we need an allocation is for a external struct e.g. `$v = (struct task_struct *)arg1; @ = *$v;`. 

Most cases we can reuse existing BPF memory and not create a new allocation.
"

I wouldn't change your 2nd paragraph "Note this function does NOT determine..."

Reply:
Fixed!